src/planning/recipe-index.ts

```diff
@@ -308,10 +315,16 @@ export class RecipeIndex {
    * - `matchingHandleConn` - a handle connection of a particle, whose slot connection is explored
    *    as a potential match to a slot above.
    */
-  private _fatesAndDirectionsMatch(slotHandleConn, matchingHandleConn): boolean {
+  private _fatesAndDirectionsMatch(slotHandleConn: HandleConnection, matchingHandleConn: HandleConnection): boolean {
     const matchingHandle = matchingHandleConn.handle;
     const allMatchingHandleConns = matchingHandle ? matchingHandle.connections : [matchingHandleConn];
-    const matchingHandleConnsHasOutput = allMatchingHandleConns.find(conn => ['out', 'inout'].includes(conn.direction));
+    const matchingDirection: Direction[] = ['out', 'inout'];
+    const matchingHandleConnsHasOutput = allMatchingHandleConns.find(conn => conn.direction && matchingDirection.includes(conn.direction));
```

Comment:
what's the reason for this change? i don't think that this is more readable.
also, HandleConnection has a `isOutput()` getter method, maybe it can be used here instead of string matching?

Reply:
`['out', 'inout']` is untyped in the find predicate, also conn.direction could be undefined.

Switched check for `isOutput()` which is much more readable.  Thanks!